Library/Homebrew/cmd/upgrade.rb

```diff
@@ -171,4 +177,35 @@ def upgrade_formula(f)
       nil
     end
   end
+
+  def relink_dependents(formula)
+    broken_formulae = []
+
+    CacheStoreDatabase.use(:linkage) do |db|
+      kegs = Formula.installed
+                    .collect(&:opt_or_installed_prefix_keg)
+                    .reject { |f| f.runtime_dependencies.to_a.empty? }
+
+      dependents = kegs.keep_if do |f|
+        f.runtime_dependencies
+         .any? { |d| d["full_name"] == formula.full_name }
+      end
+
+      break if dependents.empty?
+
+      ohai "Checking dependents of #{formula} for broken dynamic library links" if ARGV.verbose?
+
+      dependents.each do |keg|
+        checker = LinkageChecker.new(keg, cache_db: db)
+        broken_formulae << Formulary.from_keg(keg) if checker.broken_library_linkage?
+      end
+    end
+
+    return if broken_formulae.empty?
+
+    broken_formulae.each do |f|
+      ohai "#{formula}'s dependent #{f} has broken dynamic library links!"
+      reinstall_formula(f, build_from_source: true)
```

Comment:
What happens if one of these fail to install?

Reply:
Just as in `reinstall.rb`, the error bubbles up the call stack. Should I do something like this (extracted from `upgrade`)?
```ruby
    broken_formulae.each do |f|
      ohai "#{formula}'s dependent #{f} has broken dynamic library links!"
      reinstall_formula(f, build_from_source: true)
    rescue FormulaInstallationAlreadyAttemptedError
      # We already attempted to reinstall f as part of the dependency tree of
      # another formula. In that case, don't generate an error, just move on.
      nil
    rescue CannotInstallFormulaError => e
      ofail e
    rescue BuildError => e
      e.dump
      puts
      Homebrew.failed = true
    rescue DownloadError => e
      ofail e
    end
```

Reply:
Yeh, I think something like that would be ideal 👍. We want to ensure this will continue to work through formulae that fail to install. Additionally it might be nice to have a message printing the list of all formulae that we'll attempt to reinstall before we do so 👍